Bonsai.Ximea/XimeaCapture.cs

```diff
@@ -0,0 +1,99 @@
+﻿using OpenCV.Net;
+using System;
+using System.ComponentModel;
+using System.Reactive;
+using System.Reactive.Linq;
+using System.Threading.Tasks;
+using xiApi;
+using xiApi.NET;
+
+namespace Bonsai.Ximea
+{
+    [Description("Acquires a sequence of images from a Ximea camera.")]
+    public class XimeaCapture : Source<XimeaDataFrame>
+    {
+        static readonly object systemLock = new object();
+
+        [TypeConverter(typeof(SerialNumberConverter))]
+        [Description("The serial number of the camera from which to acquire images.")]
+        public string SerialNumber { get; set; }
+
+        [Description("The timeout duration, in milliseconds, when waiting for new frames to arrive.")]
+        public int Timeout { get; set; } = 1000000;
+
+        [Description("Exposure time of each frame, in microseconds.")]
+        public int Exposure { get; set; } = 2000;
+
+        [Description("Gain configuration, in dB.")]
+        public float Gain { get; set; } = 5;
```

Comment:
Does XIMEA provide any options for loading a config file? For the other packages we tend to prefer keeping all configurations external by default.

Keeping the virtual `Configure` method makes sense for derived packages for specific applications, but it's always hard to decide which exact parameters deserve to be available, e.g. why just `Exposure` and `Gain` and not trigger mode or other options? It's a slippery slope, so might as well just take the defaults from the config app or ideally from a config file.

Reply:
I tried to look for something like this but i did not find it in their API, I will take another look. The decision of what parameters to include was made by replicating the choice for the current spinnaker package.

Reply:
Unfortunately, it does seem like they dont have a method to import settings. The alternative is to create our own format, but I don't think it is worth the trouble, I would rather just have an example of how to expose the properties via overloading the virtual method.